### ISSUE TITLE
Make on CentOS Stream 9 fails

### ISSUE THREAD

**Issue by AleksovAnry** [SCST-project/scst#167]
When i try to make rpm or rpm-dkms got this error:

```
[root@c scst]# make rpm-dkms
make scst-dkms-rpm
make[1]: Entering directory '/root/scst'
name=scst-dkms &&						\
rpmtopdir=/usr/src/packages &&					\
make scst-dist-gzip &&					\
for d in BUILD RPMS SOURCES SPECS SRPMS; do			\
  mkdir -p ${rpmtopdir}/$d;					\
done &&								\
cp scst-3.8.0.e2a6774a.tar.bz2 ${rpmtopdir}/SOURCES &&		\
sed "s/@rpm_version@/3.8.0.e2a6774a/g"				\
	<${name}.spec.in >${name}.spec &&			\
MAKE="make" rpmbuild --define="%_topdir ${rpmtopdir}"	\
    --define="%kversion 5.14.0-319.el9.x86_64"			\
    --define="%kdir /lib/modules/5.14.0-319.el9.x86_64/build"			\
    --define="%pkg_build_mode 2release"		\
    -ba ${name}.spec &&					\
rm -f scst-3.8.0.e2a6774a.tar.bz2
make[2]: Entering directory '/root/scst'
make[2]: Nothing to be done for 'scst-dist-gzip'.
make[2]: Leaving directory '/root/scst'
kversion=5.14.0-319.el9.x86_64

kernel_rpm=kernel-modules-core

krpmver=5.14.0-319.el9

kernel_devel_rpm=kernel-modules-core-devel

setting SOURCE_DATE_EPOCH=1583625600
error: Failed build dependencies:
	kernel-modules-core-devel = 5.14.0-319.el9 is needed by scst-dkms-3.8.0.e2a6774a-1.el9.x86_64
make[1]: *** [Makefile:366: scst-dkms-rpm] Error 11
make[1]: Leaving directory '/root/scst'
make: *** [Makefile:392: rpm-dkms] Error 2
```

**Comment by lnocturno** [comment]
Hi,

Thank you for the report!

Thanks,
Gleb

**Comment by AleksovAnry** [comment]
Tank you for this fix, but Still cannot make. Next issue:

```
[root@c ~]#     git clone https://github.com/SCST-project/scst.git
Cloning into 'scst'...
remote: Enumerating objects: 64863, done.
remote: Counting objects: 100% (580/580), done.
remote: Compressing objects: 100% (229/229), done.
remote: Total 64863 (delta 385), reused 447 (delta 342), pack-reused 64283
Receiving objects: 100% (64863/64863), 27.68 MiB | 10.54 MiB/s, done.
Resolving deltas: 100% (51404/51404), done.
[root@c ~]# cd scst/
[root@c scst]# make rpm-dkms
make scst-dkms-rpm
make[1]: Entering directory '/root/scst'
name=scst-dkms &&						\
rpmtopdir=/usr/src/packages &&					\
make scst-dist-gzip &&					\
for d in BUILD RPMS SOURCES SPECS SRPMS; do			\
  mkdir -p ${rpmtopdir}/$d;					\
done &&								\
cp scst-3.8.0.1756a543.tar.bz2 ${rpmtopdir}/SOURCES &&		\
sed "s/@rpm_version@/3.8.0.1756a543/g"				\
	<${name}.spec.in >${name}.spec &&			\
MAKE="make" rpmbuild --define="%_topdir ${rpmtopdir}"	\
    --define="%kversion 5.14.0-319.el9.x86_64"			\
    --define="%kdir /lib/modules/5.14.0-319.el9.x86_64/build"			\
    --define="%pkg_build_mode 2release"		\
    -ba ${name}.spec &&					\
rm -f scst-3.8.0.1756a543.tar.bz2
make[2]: Entering directory '/root/scst'
[ -n "j" ] && [ -n "bz2" ] && [ -n "3.8.0.1756a543" ] && [ -n "grep -E '^debian/|^doc/|^fcst/|^iscsi-scst/|^Makefile|^qla2x00t(|-32gbit)/|^scripts/|^scst.spec|^scst/|^scst_local/|^srpt/|^usr/|^scstadmin/'" ] && name=scst && mkdir "${name}-3.8.0.1756a543" && { { scripts/list-source-files | grep -v '/\.gitignore' && if [ -e debian/changelog ]; then echo debian/changelog; fi; if [ -e debian/compat ]; then echo debian/compat; fi; } | grep -E '^debian/|^doc/|^fcst/|^iscsi-scst/|^Makefile|^qla2x00t(|-32gbit)/|^scripts/|^scst.spec|^scst/|^scst_local/|^srpt/|^usr/|^scstadmin/' | tar -T- -cf- | tar -C "${name}-3.8.0.1756a543" -xf-; } && rm -f "${name}-3.8.0.1756a543.tar.bz2" && tar -cj -f "${name}-3.8.0.1756a543.tar.bz2" "${name}-3.8.0.1756a543" && rm -rf "${name}-3.8.0.1756a543"
make[2]: Leaving directory '/root/scst'
kversion=5.14.0-319.el9.x86_64

kernel_rpm=kernel-modules-core

krpmver=5.14.0-319.el9

kernel_devel_rpm=kernel-devel

setting SOURCE_DATE_EPOCH=1583625600
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.s46STL
+ umask 022
+ cd /usr/src/packages/BUILD
+ cd /usr/src/packages/BUILD
+ rm -rf scst-3.8.0.1756a543
+ /usr/bin/bzip2 -dc /usr/src/packages/SOURCES/scst-3.8.0.1756a543.tar.bz2
+ /usr/bin/tar -xof -
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd scst-3.8.0.1756a543
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ RPM_EC=0
++ jobs -p
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.Rt0NRn
+ umask 022
+ cd /usr/src/packages/BUILD
+ cd scst-3.8.0.1756a543
+ export KVER=5.14.0-319.el9.x86_64 PREFIX=/usr
+ KVER=5.14.0-319.el9.x86_64
+ PREFIX=/usr
+ export BUILD_2X_MODULE=y CONFIG_SCSI_QLA_FC=y CONFIG_SCSI_QLA2XXX_TARGET=y
+ BUILD_2X_MODULE=y
+ CONFIG_SCSI_QLA_FC=y
+ CONFIG_SCSI_QLA2XXX_TARGET=y
+ make 2release
make[2]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543'
cd scst && make 2release
make[3]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst'
make[3]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst'
make[2]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543'
+ for d in scst fcst iscsi-scst qla2x00t-32gbit/qla2x00-target scst_local srpt
+ make -C scst
make[2]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst'
cd src && make all
make[3]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src'
echo "/* Autogenerated, don't edit */" >../include/scst_itf_ver.h
echo "" >>../include/scst_itf_ver.h
echo -n "#define SCST_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
echo -n "#define SCST_CONST_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst_const.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
echo -n "#define DEV_USER_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst_user.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
make -C certs KDIR=/lib/modules/5.14.0-319.el9.x86_64/build					\
  
make[4]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/certs'
openssl req -new -nodes -utf8 -sha512 -days 365000 \
	-batch -x509 -config x509.genkey -outform DER -out scst_module_key.der \
	-keyout scst_module_key.priv
.+...+...+.....+............+....+.........+..+....+...+...+.....+...+.+........+............+....+.....+.+...+.....+...+...+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*....+.+..+.......+..+.+..+......+...............+.......+.........+...+..+....+......+...+.....+.............+..+.+......+...........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+......+..+.......+..............+......+...+................+.........+.....+......+....+...........+.+.....+.........+....+..+..........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
....+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+...+......+..+...+..........+...........+......+..................+......+.+...+.........+......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..............+......+.....+.+......+...+.....+.+.........+..+.......+..+......+.......+...+...+.....................+.....+............................+...........+.+...........+............+.......+..............+......+...+............+....+.....+.+......+..+.+...........+..........+.....+.+.....+....+......+..............+.+........+......+...............+...+.+......+......+........+.......+..+..................+.........+..........+..............+....+..+....+.................+...................+...+...+...+..............+.+..+....+...+........+...+......+.....................+............+..........+........+...+.....................+.........+......+......+...+.+......+...+..........................+..........+...+........+....+...+...........+.+........+.+..+.......+........+...............+...+..........+..+.......+......+............+.....+..........+..+............+.........+.+...........+...+....+...+........+................+.........+..+.............+..+...+.+.....+...................+..+............+......+....+...+..............+.+...+..+.........+...+.+..+............+............+.........+.+........+......+....+........+....+.....+.........................+...+..+....+..+.......+...+...+...........+.........+.........+............+...+.......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
chmod 600 scst_module_key.der
make[4]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/certs'
make -C /lib/modules/5.14.0-319.el9.x86_64/build M=/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src				\
  
make[4]: Entering directory '/usr/src/kernels/5.14.0-319.el9.x86_64'
  CC [M]  /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/scst_copy_mgr.o
In file included from /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/../include/scst.h:64,
                 from /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/scst_copy_mgr.c:17:
/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/../include/backport.h: In function 'register_shrinker_backport':
/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/../include/backport.h:813:16: error: too few arguments to function 'register_shrinker'
  813 |         return register_shrinker(shrinker);
      |                ^~~~~~~~~~~~~~~~~
In file included from ./include/linux/list_lru.h:13,
                 from ./include/linux/fs.h:13,
                 from ./include/linux/highmem.h:5,
                 from ./include/linux/bvec.h:10,
                 from ./include/linux/blk_types.h:10,
                 from ./include/linux/blkdev.h:9,
                 from /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/../include/scst.h:37,
                 from /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/scst_copy_mgr.c:17:
./include/linux/shrinker.h:103:27: note: declared here
  103 | extern int __printf(2, 3) register_shrinker(struct shrinker *shrinker,
      |                           ^~~~~~~~~~~~~~~~~
make[5]: *** [scripts/Makefile.build:297: /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src/scst_copy_mgr.o] Error 1
make[4]: *** [Makefile:1924: /usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src] Error 2
make[4]: Leaving directory '/usr/src/kernels/5.14.0-319.el9.x86_64'
make[3]: *** [Makefile:75: all] Error 2
make[3]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst/src'
make[2]: *** [Makefile:36: all] Error 2
make[2]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.1756a543/scst'
error: Bad exit status from /var/tmp/rpm-tmp.Rt0NRn (%build)


RPM build errors:
    Bad exit status from /var/tmp/rpm-tmp.Rt0NRn (%build)
make[1]: *** [Makefile:366: scst-dkms-rpm] Error 1
make[1]: Leaving directory '/root/scst'
make: *** [Makefile:392: rpm-dkms] Error 2
```

**Comment by lnocturno** [comment]
Hi,

Fix candidate has been merged into the master branch, so you can try again.

Thanks,
Gleb

**Comment by AleksovAnry** [comment]
Hello! Again got an error while make

`
make rpm-dkms
make scst-dkms-rpm
make[1]: Entering directory «/root/scst»
name=scst-dkms &&						\
rpmtopdir=/usr/src/packages &&					\
make scst-dist-gzip &&					\
for d in BUILD RPMS SOURCES SPECS SRPMS; do			\
  mkdir -p ${rpmtopdir}/$d;					\
done &&								\
cp scst-3.8.0.3c8f66e2.tar.bz2 ${rpmtopdir}/SOURCES &&		\
sed "s/@rpm_version@/3.8.0.3c8f66e2/g"				\
	<${name}.spec.in >${name}.spec &&			\
MAKE="make" rpmbuild --define="%_topdir ${rpmtopdir}"	\
    --define="%kversion 5.14.0-362.el9.x86_64"			\
    --define="%kdir /lib/modules/5.14.0-362.el9.x86_64/build"			\
    --define="%pkg_build_mode 2release"		\
    -ba ${name}.spec &&					\
rm -f scst-3.8.0.3c8f66e2.tar.bz2
make[2]: вход в каталог «/root/scst»
make[2]: Цель «scst-dist-gzip» не требует выполнения команд.
make[2]: выход из каталога «/root/scst»
kversion=5.14.0-362.el9.x86_64

kernel_rpm=kernel-modules-core

krpmver=5.14.0-362.el9

kernel_devel_rpm=kernel-devel

setting SOURCE_DATE_EPOCH=1583625600
Executing(%prep): /bin/sh -e /var/tmp/rpm-tmp.8DNMR4
+ umask 022
+ cd /usr/src/packages/BUILD
+ cd /usr/src/packages/BUILD
+ rm -rf scst-3.8.0.3c8f66e2
+ /usr/bin/tar -xof -
+ /usr/bin/bzip2 -dc /usr/src/packages/SOURCES/scst-3.8.0.3c8f66e2.tar.bz2
+ STATUS=0
+ '[' 0 -ne 0 ']'
+ cd scst-3.8.0.3c8f66e2
+ /usr/bin/chmod -Rf a+rX,u+w,g-w,o-w .
+ RPM_EC=0
++ jobs -p
+ exit 0
Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.udwKy6
+ umask 022
+ cd /usr/src/packages/BUILD
+ cd scst-3.8.0.3c8f66e2
+ export KVER=5.14.0-362.el9.x86_64 PREFIX=/usr
+ KVER=5.14.0-362.el9.x86_64
+ PREFIX=/usr
+ export BUILD_2X_MODULE=y CONFIG_SCSI_QLA_FC=y CONFIG_SCSI_QLA2XXX_TARGET=y
+ BUILD_2X_MODULE=y
+ CONFIG_SCSI_QLA_FC=y
+ CONFIG_SCSI_QLA2XXX_TARGET=y
+ make 2release
make[2]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2'
cd scst && make 2release
make[3]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst'
make[3]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst'
make[2]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2'
+ for d in scst fcst iscsi-scst qla2x00t-32gbit/qla2x00-target scst_local srpt
+ make -C scst
make[2]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst'
cd src && make all
make[3]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src'
echo "/* Autogenerated, don't edit */" >../include/scst_itf_ver.h
echo "" >>../include/scst_itf_ver.h
echo -n "#define SCST_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
echo -n "#define SCST_CONST_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst_const.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
echo -n "#define DEV_USER_INTF_VER " >>../include/scst_itf_ver.h
echo "\"`sha1sum ../include/scst_user.h|awk '{printf $1}'`\"" >>../include/scst_itf_ver.h
make -C certs KDIR=/lib/modules/5.14.0-362.el9.x86_64/build					\
  
make[4]: Entering directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/certs'
openssl req -new -nodes -utf8 -sha512 -days 365000 \
	-batch -x509 -config x509.genkey -outform DER -out scst_module_key.der \
	-keyout scst_module_key.priv
...+.+.....+....+...+...+....................+.........+......+....+.....+...+..........+...+..+..........+..+.+..+.........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.+...+...........+....+.................+.+..+.......+...+........+.........+....+.....+......+.+..+.+..............+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*....+.+.....+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
....+.....+.+.........+......+.....+.+......+............+..+.........+.+........+.+......+.....+...+...+...+.+...+..+......+.......+...+..............+......+...+....+...+...+......+.....+....+...........+.+..+.+.....+....+..+......+...+......+.............+...+...+.........+.....+...+..........+..+......+.......+..+......+.+......+.........+.....+...+...+.......+...+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*............+..+.+.....+...+....+...+..+.+...........+..........+...+......+......+.....+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*...+......+.....+......+.+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
-----
chmod 600 scst_module_key.der
make[4]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/certs'
make -C /lib/modules/5.14.0-362.el9.x86_64/build M=/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src				\
  
make[4]: Entering directory '/usr/src/kernels/5.14.0-362.el9.x86_64'
  CC [M]  /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/scst_copy_mgr.o
In file included from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/scst.h:67,
                 from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/scst_copy_mgr.c:17:
/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/backport.h: In function 'fc_host_fpin_rcv_backport':
/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/backport.h:1781:16: error: too few arguments to function 'fc_host_fpin_rcv'
 1781 |         return fc_host_fpin_rcv(shost, fpin_len, fpin_buf);
      |                ^~~~~~~~~~~~~~~~
In file included from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/backport.h:50,
                 from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/scst.h:67,
                 from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/scst_copy_mgr.c:17:
./include/scsi/scsi_transport_fc.h:860:6: note: declared here
  860 | void fc_host_fpin_rcv(struct Scsi_Host *shost, u32 fpin_len, char *fpin_buf,
      |      ^~~~~~~~~~~~~~~~
In file included from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/scst.h:67,
                 from /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/scst_copy_mgr.c:17:
/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/backport.h:1781:16: error: 'return' with a value, in function returning void [-Werror=return-type]
 1781 |         return fc_host_fpin_rcv(shost, fpin_len, fpin_buf);
      |                ^~~~~~~~~~~~~~~~
/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/../include/backport.h:1769:1: note: declared here
 1769 | fc_host_fpin_rcv_backport(struct Scsi_Host *shost, u32 fpin_len, char *fpin_buf,
      | ^~~~~~~~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:299: /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src/scst_copy_mgr.o] Error 1
make[4]: *** [Makefile:1928: /usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src] Error 2
make[4]: Leaving directory '/usr/src/kernels/5.14.0-362.el9.x86_64'
make[3]: *** [Makefile:83: all] Error 2
make[3]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst/src'
make[2]: *** [Makefile:36: all] Error 2
make[2]: Leaving directory '/usr/src/packages/BUILD/scst-3.8.0.3c8f66e2/scst'
`
